services/api/src/models/billing.ts

```diff
@@ -0,0 +1,202 @@
+import moment from 'moment';
+import { Group, BillingGroup } from './group';
+import { getKeycloakAdminClient } from '../clients/keycloak-admin';
+import {
+  handleAddBillingModifier,
+  handleGetBillingGroupModifiers,
+  handleUpdateBillingModifier,
+  handleDeleteBillingModifier,
+  handleDeleteAllBillingGroupModifier
+} from '../resources/billing/helpers';
+
+import { GroupInput } from './group';
+
+export interface BillingModifier {
+  id?: number;
+  group?: Group | GroupInput;
+  startDate?: string;
+  endDate?: string;
+  discountFixed?: number;
+  discountPercentage?: number;
+  extraFixed?: number;
+  extraPercentage?: number;
+  customerComments?: string;
+  adminComments?: string;
+  weight?: number;
+}
+
+export interface BillingModifierInput extends BillingModifier {
+  group: GroupInput;
+}
+
+export interface AddBillingModifierInput {
+  input: BillingModifierInput;
+}
+
+export interface UpdateBillingModifierInput {
+  input: {
+    id: number;
+    patch: BillingModifier;
+  };
+}
+
+export interface BillingModifiersInput {
+  input: GroupInput;
+  month?: string;
+}
+
+export interface DeleteBillingModifierInput {
+  input: {
+    id: number;
+  };
+}
+
+export interface DeleteAllBillingGroupModifiersInput {
+  input: GroupInput;
+}
+
+const loadGroup = async (group: GroupInput) => {
+  const keycloakAdminClient = await getKeycloakAdminClient();
+  const GroupModel = Group({ keycloakAdminClient });
+  return GroupModel.loadGroupByIdOrName(group);
+};
+
+export const convertDateToMYSQLDateTimeFormat = (date: string) => {
+  const DATE_FORMAT = 'YYYY-MM-DD HH:mm:ss';
+  return moment(new Date(date).toISOString())
+    .format(DATE_FORMAT)
+    .toString();
+};
+
+const prepareAddBillingModifierFromInput = async (
+  input: BillingModifierInput
+) => {
+  const { group: groupInput, ...rest } = input;
+  const group = await loadGroup(groupInput);
+
+  const startDate = convertDateToMYSQLDateTimeFormat(rest.startDate);
+  const endDate = convertDateToMYSQLDateTimeFormat(rest.endDate);
+
+  return {
+    group,
+    billingModifier: { ...rest, gid: group.id, startDate, endDate }
+  };
+};
+
+/**
+ * Create/Add Billing Modifier
+ *
+ * @param {AddBillingModifierInput} input The modifier values
+ *
+ * @return {BillingModifier} The created modifier
+ */
+export const addBillingModifier = async (input: BillingModifierInput) => {
+  const { group, billingModifier } = await prepareAddBillingModifierFromInput(
```

Comment:
Related to previous comment: all these prepare* functions feel like they should be in the resolver, since they are mapping the API object into the expected model object.

Reply:
Addressed in 4a6a75dfaeb3cbcb6da384581a440223fd5d83cb